packages/react/src/analytics/integrations/Castle/types/index.ts

```diff
@@ -0,0 +1,37 @@
+import type { AxiosInstance } from 'axios';
+import type {
+  configure,
+  createRequestToken,
+  custom,
+  form,
+  page,
+} from '@castleio/castle-js';
+import type { IntegrationOptions } from '@farfetch/blackout-analytics';
+
+export interface CastleIntegrationOptions extends IntegrationOptions {
```

Comment:
Since `CastleIntegrationOptions` is baed on the ConfigureOptions from `castle` package, this could be done instead:

```suggestion

import type {
  ConfigureOptions,
} from '@castleio/castle-js';

export interface CastleIntegrationOptions extends IntegrationOptions {
configureOptions: ConfigureOptions
debugModeOn?: boolean;
  httpClient?: AxiosInstance;
  clientIdHeaderName?: string;
  configureHttpClient?: (
    tokenCreator: typeof createRequestToken,
  ) => Promise<void>;
```

If a separate property is used for the configure options, if there is a new version of '@castleio/castle-js' package that adds new properties, the integration code will not need to be changed. However for this to work, you will need to change this call as well:

```js
applyOptions(options: CastleIntegrationOptions) {
    try {
      this.castleJS.configure(options.configureOptions); <--- Just pass the configureOptions prop directly into the method

      this.installInterceptor(options);
    } catch (error) {
      throw new Error(
        `${CASTLE_MESSAGE_PREFIX} Failed to initialize Castle. Error: ${error}`,
      );
    }
```
What do you think?